File: viralt-app/src/types/referral.ts
export interface ReferralInfo {
  referralId: number;
  campaignId: number;
  referrerClientId: number;
  referredClientId: number;
  createdAt: string;
  bonusEntriesAwarded: number;
  referrerName?: string;
  referredName?: string;
}

File: Viralt.Domain/Models/Referral.cs
namespace Viralt.Domain.Models;

public class Referral
{
    public long ReferralId { get; set; }
    public long CampaignId { get; set; }
    public long ReferrerClientId { get; set; }
    public long ReferredClientId { get; set; }
    public DateTime CreatedAt { get; set; }
    public int BonusEntriesAwarded { get; set; }

    public virtual Campaign Campaign { get; set; }
    public virtual Client ReferrerClient { get; set; }
    public virtual Client ReferredClient { get; set; }
}
